language_load.py

```diff
@@ -125,10 +125,16 @@ def update_translation(target_lang, ref_lang, verbose, filename, strings_by_obje
         previously_translated = translation_existed(target_lang, data['strings'][string_key])
         translation_needs_update = False
         if previously_translated:
+            translation = data['strings'][string_key][target_lang]
+            target_string = strings_by_object[obj_id][string_key]
             translation_needs_update = translation_changed(verbose, obj_id, string_key,\
-                                          data['strings'][string_key][target_lang],\
-                                          strings_by_object[obj_id][string_key])
-        if not previously_translated or translation_needs_update:
+                                          translation, target_string)
+        else:
+            translation = data['strings'][string_key][ref_lang]
+            target_string = strings_by_object[obj_id][string_key]
+            translation_needs_update = translation_changed(verbose, obj_id, string_key,\
+                                          translation, target_string)
+        if translation_needs_update:
```

Comment:
For what it's worth, I don't think this fully reverts to the old behaviour? Both branches (if/else) are identical, and the `not previously_translated ` is missing here.

Reply:
Doesn't it? Do note that the `translation` object changes from `target_lang` to `ref_lang` (that is the only difference.

So the update is done if `translation_existed && translation_changed` (old translation vs new translation) or `!translation_existed && translation_changed` (ref translation vs new translation)

```python
        previously_translated = translation_existed(target_lang, data['strings'][string_key])
        translation_needs_update = False
        if previously_translated:
            translation = data['strings'][string_key][target_lang]
            target_string = strings_by_object[obj_id][string_key]
            translation_needs_update = translation_changed(verbose, obj_id, string_key,\
                                          translation, target_string)
        else:
            translation = data['strings'][string_key][ref_lang]
            target_string = strings_by_object[obj_id][string_key]
            translation_needs_update = translation_changed(verbose, obj_id, string_key,\
                                          translation, target_string)
        if translation_needs_update:
            print("Updating " + obj_id + " string '" + string_key + "'")
            data['strings'][string_key][target_lang] = strings_by_object[obj_id][string_key]
            updated = True
```

Reply:
Ah, you're right. In that case, I'd really prefer this difference to be more explicit, by only branching between either assignment to `translation`. The rest can indeed be shared.

Reply:
Perfect, I've simplified it a great deal in the last commit.